packages/mobile/src/screens/chat-screen/ChatMessageListItem.tsx

```diff
@@ -182,31 +161,46 @@ export const ChatMessageListItem = forwardRef<View, ChatMessageListItemProps>(
             styleProp
           ]}
         >
-          <TouchableWithoutFeedback onPress={onLongPress}>
-            <View
-              ref={refProp}
-              onLayout={(e) => {
-                const { width, height } = e.nativeEvent.layout
-                setReactionPosition({
-                  x: width - REACTION_OFFSET_X,
-                  y: height - REACTION_OFFSET_Y
-                })
-              }}
-            >
-              {/* This View is for measuring y offset and height to calculate
-            position of reactions popup. */}
-              <View
-                ref={refProp}
-                style={[styles.bubble, isAuthor && styles.isAuthor]}
-              >
-                <Text
-                  style={[styles.message, isAuthor && styles.messageIsAuthor]}
+          <View>
+            <TouchableWithoutFeedback onPress={onLongPress}>
+              <View>
+                <View
+                  style={[styles.bubble, isAuthor && styles.isAuthor]}
+                  ref={refProp}
                 >
-                  {message.message}
-                </Text>
+                  <Text
+                    style={[styles.message, isAuthor && styles.messageIsAuthor]}
+                  >
+                    {message.message}
+                  </Text>
+                </View>
+                {message.reactions?.length > 0 ? (
+                  <>
+                    {shouldShowReaction ? (
+                      <View
+                        style={[
+                          styles.reactionContainer,
+                          isAuthor
+                            ? styles.reactionContainerIsAuthor
+                            : styles.reactionContainerOtherUser
+                        ]}
+                      >
+                        {message.reactions.map((reaction, index) => {
+                          return (
+                            <ChatReaction key={index} reaction={reaction} />
+                          )
+                        })}
+                      </View>
+                    ) : null}
+                    {/* Add an 8px margin between messages when there are reactions. */}
+                    {!hasTail ? (
+                      <View style={styles.reactionMarginBottom} />
```

Comment:
I'd make this a style on the root view rather than having a proxy element

Reply:
ty for late & speedy reviews!